parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -423,14 +408,14 @@ async def copyfile(
         Raises:
             FSError
         """
-        await self.touch(target_path, exist_ok=exist_ok)
-        offset = 0
-        while True:
-            buff = await self.read_bytes(source_path, length, offset * length)
-            if not buff:
-                break
-            await self.write_bytes(target_path, buff, offset * length)
-            offset += 1
+        write_mode = "wb" if exist_ok else "xb"
+        async with await self.open_file(source_path, mode="rb") as source:
+            async with await self.open_file(target_path, mode=write_mode) as target:
+                while True:
+                    data = await source.read(length)
```

Comment:
`length` feels like a bad name for this parameter (it seems it there to copy only part of the file, where it is there to controle the size of the buffer for the copy)... so we should rename this `buffer_size` or even remove this parameter entirely given we don't use it in our codebase ;-)

Reply:
Configuring the `buffer_size` copy seems ok to me, I'll rename it :)